### PR TITLE
Replace Engagement View with Frontend Directory

### PR DESCRIPTION
**Pull request by andrea-grapl**
Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/1012

What changes does this PR make to Grapl? Why?
Removes engagement view and replaces with a new frontend directory using updated dependencies which also work in CI/CD. Created project using create a new react app using yarn as the testing framework, unit testing frameworks, linting, formatting, & proxy. Also added updated Make & Docker files 

How were these changes tested?
yarn build
Navigate to localhost:3000.

### CODE REVIEW

File: src/js/frontend/Makefile [hunk 1]
```diff
@@ -87,7 +87,7 @@ install-dependencies: ## Install dependencies in the build container if necessar
 build-code: install-dependencies $(FILES)
 build-code: ## Build the code if any files have changed
 	$(DOCKER_RUN) yarn build
-	$(DOCKER_RUN) yarn analyze-build
+# 	$(DOCKER_RUN) yarn analyze-build
```
Comment (colin-grapl) [inline src/js/frontend/Makefile:90]: delete

File: src/js/frontend/package.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{
+  "name": "frontend",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.14.1",
+    "@testing-library/react": "^13.0.0",
+    "@testing-library/user-event": "^13.2.1",
```
Comment (colin-grapl) [inline src/js/frontend/package.json:8]: I assume these should be dev-dependencies?

**Review by colin-grapl — APPROVED** [review]
one thing to fix please

File: src/js/frontend/package.json (hunk 2 above)
Comment (andrea-grapl) [reply src/js/frontend/package.json]: Done

**Pull request merged**